guppylang/definition/pytket_circuits.py

```diff
@@ -74,65 +74,73 @@ def parse(self, globals: Globals, sources: SourceMap) -> "ParsedPytketDef":
             func_ast, globals.with_python_scope(self.python_scope)
         )
 
+        # Compare signatures.
         # TODO: Allow arrays as arguments.
-        # Retrieve circuit signature and compare.
-        try:
-            import pytket
-
-            if isinstance(self.input_circuit, pytket.circuit.Circuit):
-                try:
-                    import tket2  # type: ignore[import-untyped, import-not-found, unused-ignore]  # noqa: F401
-
-                    qubit = cast(TypeDef, globals["qubit"]).check_instantiate(
-                        [], globals
-                    )
-
-                    circuit_signature = FunctionType(
-                        [FuncInput(qubit, InputFlags.Inout)]
-                        * self.input_circuit.n_qubits,
-                        row_to_type([bool_type()] * self.input_circuit.n_bits),
-                    )
-
-                    if not (
-                        circuit_signature.inputs == stub_signature.inputs
-                        and circuit_signature.output == stub_signature.output
-                    ):
-                        # TODO: Implement pretty-printing for signatures in order to add
-                        # a note for expected vs. actual types.
-                        raise GuppyError(PytketSignatureMismatch(func_ast, self.name))
-                except ImportError:
-                    err = Tket2NotInstalled(func_ast)
-                    err.add_sub_diagnostic(Tket2NotInstalled.InstallInstruction(None))
-                    raise GuppyError(err) from None
-        except ImportError:
-            pass
+        circuit_signature = _signature_from_circuit(
+            self.input_circuit, globals, self.defined_at
+        )
+        if not (
+            circuit_signature.inputs == stub_signature.inputs
+            and circuit_signature.output == stub_signature.output
+        ):
```

Comment:
drive-by:
```suggestion
        if circuit_signature != stub_signature:
```

Reply:
This doesn't work because they have different `input_names`.